src/replica/disk_migration/replica_disk_migrator.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "replica/replica_stub.h"
+#include "replica_disk_migrator.h"
+
+#include <dsn/dist/fmt_logging.h>
+
+namespace dsn {
+namespace replication {
+
+replica_disk_migrator::replica_disk_migrator(replica *r) : replica_base(r), _replica(r) {}
+
+replica_disk_migrator::~replica_disk_migrator() = default;
+
+// THREAD_POOL_REPLICATION
+void replica_disk_migrator::on_migrate_replica(const replica_disk_migrate_request &req,
+                                               /*out*/ replica_disk_migrate_response &resp)
+{
+    if (!check_migration_args(req, resp)) {
+        return;
+    }
+
+    _status = disk_migration_status::MOVING;
+    ddebug_replica("received replica disk migrate request({}), update status from {}=>{}",
+                   _request_msg,
+                   enum_to_string(disk_migration_status::IDLE),
+                   enum_to_string(status()));
+
+    tasking::enqueue(
+        LPC_REPLICATION_LONG_COMMON, _replica->tracker(), [=]() { migrate_replica(req); });
+}
+
+bool replica_disk_migrator::check_migration_args(const replica_disk_migrate_request &req,
+                                                 /*out*/ replica_disk_migrate_response &resp)
+{
+    _replica->_checker.only_one_thread_access();
+
+    _request_msg = fmt::format(
+        "gpid={}, origin={}, target={}", req.pid.to_string(), req.origin_disk, req.target_disk);
```

Comment:
`replica_disk_migrator` inherit `replica_base` class, so you can use marco like `derror_replica`, the log will add replica name for each log, as a result, it is not necessary to print pid while printing request, `_request_msg` is not necessary. 
I recommend removing this variety, and update log like:
`derror_replica("received replica disk migrate request(origin={} target={}), err = {}", req.origin_disk, req.target_disk, err_msg);`.